src/tools/clippy/clippy_utils/src/ast_utils.rs

```diff
@@ -362,11 +362,23 @@ pub fn eq_variant_data(l: &VariantData, r: &VariantData) -> bool {
 }
 
 pub fn eq_struct_field(l: &FieldDef, r: &FieldDef) -> bool {
-    l.is_placeholder == r.is_placeholder
-        && over(&l.attrs, &r.attrs, |l, r| eq_attr(l, r))
-        && eq_vis(&l.vis, &r.vis)
-        && both(&l.ident, &r.ident, |l, r| eq_id(*l, *r))
-        && eq_ty(&l.ty, &r.ty)
+    if l.is_placeholder != r.is_placeholder
+    || !over(&l.attrs, &r.attrs, |l, r| eq_attr(l, r))
+    || !eq_vis(&l.vis, &r.vis)
+    {
+        false
+    } else {
+        match (&l.variant, &r.variant) {
+            (FieldVariant::Named(NamedField{ident: l_ident, ty: l_ty}),
+            FieldVariant::Named(NamedField{ident: r_ident, ty: r_ty})) =>
+            both(l_ident, r_ident, |l, r| eq_id(*l, *r))
+            && eq_ty(l_ty, r_ty),
+            // FIXME: Compare two unnamed fields and check for equality
+            (FieldVariant::Unnamed(_),
+            FieldVariant::Named(_)) => false,
```

Comment:
This should probably be 

```suggestion
            // FIXME: Compare two unnamed fields and check for equality
            (FieldVariant::Unnamed(_),
            FieldVariant::Unnamed(_)) => false,
```

and should be addressed in this PR. I think you know better how to deal / compare unnamed field than any Clippy devs. Same goes for the other ``FIXME``s, but I guess we could address them on our own. We'd prefer if you could address those directly in this PR though.

Reply:
Ok, let me implement that. For the other `FIXME`, my plan was to implement just the basic parsing, get someone to review it to make sure I'm good, and then implement the feature. Though, I will need some more mentoring to know how exactly I would tackle this problem, so I may take a bit more time.